cmd/ocagent/main.go

```diff
@@ -157,10 +169,32 @@ func runOCReceiver(addr string, sr receiver.TraceReceiverSink) (doneFn func() er
 		return nil, fmt.Errorf("Failed to register ocgrpc.DefaultServerViews: %v", err)
 	}
 
-	if err := ocr.StartTraceReception(context.Background(), sr); err != nil {
-		return nil, fmt.Errorf("Failed to start TraceReceiver: %v", err)
+	ctx := context.Background()
+
+	switch {
+	case acfg.CanRunAllOpenCensusReceivers():
```

Comment:
Why have the CanRunAllOpenCensusReceivers as a separate function? At first glance the only thing different in this case is the call to ocr.Stop if starting metrics fail and traces were started. It seems that if statements will avoid the the very similar code between the cases. The single log message can also be handled (although I don't think it is worth the trouble).

Reply:
Aha you read my mind, I now see it isn't worth it making CanRunAllOpenCensusReceivers, I'll revert that, thank you!